### PR TITLE
Include tests in sdist tarball

### PR DESCRIPTION
**Pull request by floppym**
Fixes: https://github.com/kennethreitz/requests/issues/4024

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1 +1,2 @@
 include README.rst LICENSE NOTICE HISTORY.rst requests/cacert.pem
+recursive-include tests *.py
```
Comment (nateprewitt) [inline MANIFEST.in:2]: Perhaps `graft tests` would be a more permissive command to use here. That way we're capturing possibly required files in the tests directory that aren't necessarily python files.

Comment (floppym) [reply MANIFEST.in:2]: Using graft causes any `pyc` files to also be picked up -- we don't want that.

Comment (nateprewitt) [reply MANIFEST.in:2]: If we did a release with dirty directories, you're right, that could be a problem. I'd suggest using something like `global-exclude *.py[co]` to solve the `.pyc` issue. That approach will be more inline with most of the other major python projects.

Comment (floppym) [reply MANIFEST.in:2]: I created #4027 with graft/exclude-global. I leave it to up to you guys to pick one. :)

Comment (nateprewitt) [reply MANIFEST.in:2]: Thanks @floppym :)

**Comment by Lukasa** [comment]
I'd rather take the recursive-include for now.

**Comment by nateprewitt** [comment]
Given that this was overlooked for a year, it seems like an easy place to forget about again. Perhaps simply changing it to `recursive-include tests *` which has worked for `urllib3`? You handle the releases though, so whatever you think works best, I'm on board with.

As for the failure, it's due to our Sphinx dependency in dev-packages. A release that came out today finally broke 2.6 which Sphinx stopped supporting a while ago. We'll likely want to pin it at `<=1.5.5` until we deprecate 2.6 support ourselves.

**Comment by sigmavirus24** [comment]
> Given that this was overlooked for a year, it seems like an easy place to forget about again.

It's especially easy given that tests aren't part of the actual package and should only be included in source archives, not source distributions. But *shrug* I'm not about to have this disagreement with another downstream redistributor.

I'd also like to avoid capping Sphinx to preserve 2.6 support. I'd rather Pipfile joined the time-period it was born into and understood dependencies that were for specific versions of Python or specific environments.

**Comment by floppym** [comment]
> It's especially easy given that tests aren't part of the actual package and should only be included in source archives, not source distributions.

What's the difference between a "source archive" and a "source distribution"? I'm missing your point.

My goal is to be able to run the tests as part of the Gentoo Linux build script for this package. We skip this currently; otherwise I would have noticed the missing files sooner.

**Comment by sigmavirus24** [comment]
> What's the difference between a "source archive" and a "source distribution"? I'm missing your point.

One (a source archive) is literally the archive of the repository made by setuptools and uploaded to PyPI. The other (a source distribution) is meant to actually be installed by things Python tooling for an end user. One is meant as historical data, the other as an actual installable product (for lack of a better term). One can be downloaded by downstream redistributors and used to run tests, the other is meant to be downloaded and used by users.

> My goal is to be able to run the tests as part of the Gentoo Linux build script for this package. We skip this currently; otherwise I would have noticed the missing files sooner.

And that's an admirable goal. No one wants a downstream redistributor to package something incorrectly and only find out about it from users.

**Comment by Lukasa** [comment]
@sigmavirus24 I'm sympathetic to wanting the improvement from pipfile, but adding that feature will likely take longer than us just pinning sphinx. We can remove the pin when pipfile improves. Does that sound reasonable?

**Comment by kennethreitz** [comment]
+1 to pinning

**Comment by sigmavirus24** [comment]
If we want to pin, let's pin. At the moment we're capping. I'm fine with pinning *temporarily*.

**Comment by sigmavirus24** [comment]
Closing and re-opening to trigger new builds with our fixed dependencies

**Pull request merged**